### DISCUSSION TITLE
ldt compile error

### DISCUSSION THREAD

**Discussion by jachungithub** [NASA-LIS/LISF#1020] [Q&A] [answered]
Hi, 

I have an error message while LDT (***LISF_public_release_7.3.3***)was being compiled. 
> The ***error message*** is below:

readGLASSlaiObs.F90(95): #error: can't find include file: hdf.f90
make: *** [readGLASSlaiObs.o] Error 1

[ERR] Compile failed

> My ***configure.ldt*** is below:

FC              = mpiifort
FC77            = mpiifort
LD              = mpiifort
CC              = mpiicc
AR              = ar
MOD_ESMF        = /opt/kma/kma_lib/apps/esmf_mpi/710R/INTEL/210/mod/modO/Linux.intel.64.intelmpi.default
LIB_ESMF        = /opt/kma/kma_lib/apps/esmf_mpi/710R/INTEL/210/lib/libO/Linux.intel.64.intelmpi.default
INC_JPEG2000      = /opt/kma/kma_lib/apps/openjpeg/2.3.1/include/
LIB_JPEG2000      = /opt/kma/kma_lib/apps/openjpeg/2.3.1/lib/
INC_ECCODES     = /opt/kma/kma_lib/apps/eccodes/2.16.0/INTEL/210/include/
LIB_ECCODES     = /opt/kma/kma_lib/apps/eccodes/2.16.0/INTEL/210/lib/
INC_NETCDF      = /opt/kma/kma_lib/apps/netcdf/4.7.3/INTEL/210/include/
LIB_NETCDF      = /opt/kma/kma_lib/apps/netcdf/4.7.3/INTEL/210/lib/
INC_HDF4        = /opt/kma/kma_lib/apps/hdf4/4.2.15/INTEL/210/include/
LIB_HDF4        = /opt/kma/kma_lib/apps/hdf4/4.2.15/INTEL/210/lib/
INC_HDF5        = /opt/kma/kma_lib/apps/hdf5/1.10.3/INTEL/210/include/
LIB_HDF5        = /opt/kma/kma_lib/apps/hdf5/1.10.3/INTEL/210/lib/
INC_HDFEOS      = /opt/kma/kma_lib/apps/hdfeos2/2.20/INTEL/210/include/
LIB_HDFEOS      = /opt/kma/kma_lib/apps/hdfeos2/2.20/INTEL/210/lib/
INC_FORTRANGIS1 = /opt/kma/kma_lib/apps/fortrangis/2.5/INTEL/210/include/
INC_FORTRANGIS2 = /opt/kma/kma_lib/apps/fortrangis/2.5/INTEL/210/include/
LIB_FORTRANGIS  = /opt/kma/kma_lib/apps/fortrangis/2.5/INTEL/210/lib/
INC_LIBGEOTIFF  = /opt/kma/kma_lib/apps/geotiff/1.7.0/include/
LIB_LIBGEOTIFF  = /opt/kma/kma_lib/apps/geotiff/1.7.0/lib/
LIB_GDAL        = /opt/kma/kma_lib/apps/gdal/3.0.4/lib/
CFLAGS          = -c  -DIFC -I$(INC_LIBGEOTIFF)
FFLAGS77        = -c -O2 -nomixed-str-len-arg -names lowercase -convert big_endian -assume byterecl  -DSPMD  -DIFC -I$(MOD_ESMF) -DUSE_INCLUDE_MPI -I$(INC_ECCODES) -I$(INC_NETCDF)  -I$(INC_HDFEOS)  -I$(INC_HDF4)  -I$(INC_HDF5)  -I$(INC_FORTRANGIS1) -I$(INC_FORTRANGIS2)
FFLAGS          =  -c -O2 -u -traceback -fpe0  -nomixed-str-len-arg -names lowercase -convert big_endian -assume byterecl  -DSPMD -DIFC -I$(MOD_ESMF) -DUSE_INCLUDE_MPI -I$(INC_ECCODES) -I$(INC_NETCDF)  -I$(INC_HDFEOS)  -I$(INC_HDF4)  -I$(INC_HDF5)  -I$(INC_FORTRANGIS1) -I$(INC_FORTRANGIS2)
LDFLAGS         =  -L$(LIB_ESMF) -lesmf -lstdc++ -limf -lm -lrt -lz -lmpi -L$(LIB_ECCODES) -leccodes_f90 -leccodes -L$(LIB_JPEG2000) -lopenjp2 -L$(LIB_NETCDF) -lnetcdff -lnetcdf -L$(LIB_HDFEOS) -lhdfeos -lGctp -L$(LIB_HDF4) -lmfhdf -ldf -ljpeg -lz  -L$(LIB_HDF5) -lhdf5_fortran -lhdf5_hl -lhdf5 -ldl -L$(LIB_FORTRANGIS) -lfortrangis -lfortranc -L$(LIB_GDAL) -lgdal -L$(LIB_LIBGEOTIFF)  -ltiff -lgeotiff -lm -lz -ljpeg 

FYI, the ***Intel Comiler version is 21.3***. 
Any help would be really appreciated.
Thank you, 
Jong Ahn Chun

**Comment by bmcandr** [comment, marked as answer]
Hi @jachungithub,

First question: does the file `hdf.f90` exist in `/opt/kma/kma_lib/apps/hdf4/4.2.15/INTEL/210/include/`?

**Reply by jachungithub** [reply]
Hi @bmcandr:

Unfortunately, there are **_no *.f90_** in the directory. They are all *.h in the directory. I looked up the other directories, but I couldn't find *.f90 at all.

Thanks
Jong

**Reply by bmcandr** [reply]
Ok, make sure that you compile HDF4 with Fortran enabled.

* If using `configure` and `make`, pass `--enable-fortran`
* If using CMake, edit HDF4options.cmake to enable Fortran (search within this file to find the Fortran related entries)

**Reply by jachungithub** [reply]
Thanks a lot. I will go ahead and request the administrators to compile HDF4 with the option.

**Reply by jachungithub** [reply]
The error has been cleared. However, new errors came up. I think I'd better open a new discussion.

**Reply by Lise-Li** [reply]
Hi, @jachungithub  have your problem solved? I met the same problem with you, everytime I configure HDF4 with Fortran enabled, I will meet the error: Type mismatch between actual argument at (1) and actual argument at (2) (CHARACTER(*)/INTEGER(4)). Have you ever met the problem? Would you mind letting me know how you configured HDF4? Thank you!
